packages/libs/deer/src/lib.rs

```diff
@@ -56,217 +53,269 @@ pub trait ArrayAccess<'de> {
     fn finish(self) -> Result<(), ArrayAccessError>;
 }
 
-// TODO: Error PR: attach the expected and received type
-// Reason: this definition is used by all other visitors, we provide fallbacks, which **currently**
-//  do not take into account the value provided. This will change with the error PR.
+// Reason: We error out on every `visit_*`, which means we do not use the value, but(!) IDEs like to
+// use the name to make autocomplete, therefore names for unused parameters are required.
 #[allow(unused_variables)]
 pub trait Visitor<'de>: Sized {
     type Value;
 
-    fn expecting_display(&self) -> String;
-
-    // TODO: this is currently completely untyped, we might want to adhere to a standard, like
-    //  JSON-Schema or OpenAPI
-    //  The problem here mainly is: which crate to use, one can use utoipa (but that has significant
-    //  overhead)  there's no real library out there that properly just provides the types
-    //  necessary.
-    fn expecting_schema(&self) -> BTreeMap<String, String> {
-        BTreeMap::new()
-    }
+    fn expecting(&self) -> Schema;
 
     fn visit_none(self) -> Result<Self::Value, VisitorError> {
-        // Err(Report::new(Self::Error::message(
-        //     "unexpected missing value",
-        // )))
-        todo!()
+        Err(Report::new(MissingError)
+            .attach(ExpectedType::new(self.expecting()))
```

Comment:
Is `attach` over `attach_printable` intended?

Reply:
Yes, `Schema` has no actual `Debug`/`Display` implementation, and even if it would likely be quite unreadable, so for the sake of avoiding additional complexity, for now, it has been omitted.